sql/importer: TestExportImportBank failed

sql/importer.TestExportImportBank [failed](https://tanzanite.cluster.engflow.com/invocations/default/f306225d-360a-4bbe-8ec8-cb228a97db16?testReportRun=27&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on master @ [3292e23e914f2b5a63e892fef1705ad476be4bd7](https://github.com/cockroachdb/cockroach/commits/3292e23e914f2b5a63e892fef1705ad476be4bd7):


```
=== RUN   TestExportImportBank
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestExportImportBank3022263666
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    exportcsv_test.go:146: -- test log scope end --
test logs left over in: outputs.zip/logTestExportImportBank3022263666
--- FAIL: TestExportImportBank (9.82s)
=== RUN   TestExportImportBank/null=NULL
    exportcsv_test.go:133: error executing query="CREATE TABLE bank2 (\n\t\tid INT PRIMARY KEY,\n\t\tbalance INT,\n\t\tpayload STRING,\n\t\tFAMILY (id, balance, payload)\n\t)" args=[]: pq: relation "defaultdb.public.bank2" already exists
    --- FAIL: TestExportImportBank/null=NULL (0.67s)
=== RUN   TestExportImportBank/null=
    exportcsv_test.go:134: error executing query="IMPORT INTO bank2 CSV DATA ($1) WITH delimiter = '|', nullif = ''" args=["userfile:/t/TestExportImportBank/null=/*"]: pq: result is ambiguous: error=[NotLeaseHolderError] proxy failed, update client information desc: r68:/Tenant/10{-/Table/104/1/10} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=10, sticky=1711523159.405364670,0], lease: repl=(n3,s3):2 seq=3 start=1711519561.447063165,0 exp=1711519567.446765215,0 pro=1711519561.446765215,0, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING != et:desc:r68:/Tenant/10{-/Table/11} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11, sticky=1711523159.405364670,0], lease:repl=(n3,s3):2 seq=3 start=1711519561.447063165,0 exp=1711519567.446765215,0 pro=1711519561.446765215,0; r68: replica not lease holder; current lease is repl=(n3,s3):2 seq=3 start=1711519561.447063165,0 exp=1711519567.446765215,0 pro=1711519561.446765215,0 [propagate] (last error: key range /Tenant/10/Table/3/1/107/2/1-/Tenant/10/Table/53/1/955020615536803841/"legacy_payload"/1915-10-07T17:53:55.639340999Z/0/NULL outside of bounds of range /Tenant/10-/Tenant/10/Table/11; suggested ranges: [desc: r68:/Tenant/10{-/Table/11} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11, sticky=1711523159.405364670,0], lease: repl=(n3,s3):2 seq=3 start=1711519561.447063165,0 exp=1711519567.446765215,0 pro=1711519561.446765215,0, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING desc: r108:/Tenant/10/Table/1{1-04/1/10} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=11], lease: repl=(n3,s3):2 seq=3 start=1711519561.447063165,0 exp=1711519567.446765215,0 pro=1711519561.446765215,0, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING])
    --- FAIL: TestExportImportBank/null= (2.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=27</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExportImportBank.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37110

--- a/pkg/sql/importer/exportcsv_test.go
+++ b/pkg/sql/importer/exportcsv_test.go
@@ -131,6 +131,7 @@ func TestExportImportBank(t *testing.T) {
 			schema := bank.FromRows(1).Tables()[0].Schema
 			exportedFiles := filepath.Join(exportDir, "*")
 			db.Exec(t, fmt.Sprintf("CREATE TABLE bank2 %s", schema))
+			defer db.Exec(t, "DROP TABLE bank2")
 			db.Exec(t, fmt.Sprintf(`IMPORT INTO bank2 CSV DATA ($1) WITH delimiter = '|'%s`, nullIf), exportedFiles)
 
 			db.CheckQueryResults(t,
@@ -140,7 +141,6 @@ func TestExportImportBank(t *testing.T) {
 				`SELECT fingerprint FROM [SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE bank2]`,
 				db.QueryStr(t, `SELECT fingerprint FROM [SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE bank]`),
 			)
-			db.Exec(t, "DROP TABLE bank2")
 		})
 	}
 }

